aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -1567,6 +1439,160 @@ Tensor min_mps(const Tensor& input_t) {
     min_max_out_mps(input_t, dim, keepdim, output_t, indices_t, "min", "min_out_mps");
 }
 
+void argmin_argmax_out_mps
+   (const Tensor& input_t,
+    c10::optional<int64_t> dim,
+    bool keepdim,
+    const Tensor& output_t,
+    string reduction_type,
```

Comment:
Please create `enum class MPSReductionType` instead of assuming that all string space is limited to just 2 values.
Also enum comparison is just one assembly instruction (and can be inlined if needed), while string comparison can take up to `O(n)`, where `n` is length of string in question
```suggestion
    ReductionType reduction_type,
```